codice/ddf
# DDF-4478 Improves filtering logic for dropdowns

#### What does this PR do?
Improves filtering logic for dropdowns in Intrigue.
Previously, filtering through a dropdown would match if any part of the filter was found in the word. This would lead to weird behavior like returning "validation" and "width" when the filter was "id".

The new behavior splits the words on camelcase, dashes, dots, and spaces then checks to see if any of the resulting words start with the filter value. This results in a slightly more intuitive behavior as shown here:
![betterdropdown1](https://user-images.githubusercontent.com/9013407/51849663-70608000-22dd-11e9-9242-4c550f4eb5e9.gif)
![betterdropdown2](https://user-images.githubusercontent.com/9013407/51849667-73f40700-22dd-11e9-9b42-a6f857b96b34.gif)

#### Who is reviewing it? 
@emmberk @ethantmanns @Kjames5269 @jhunzik 
@TeresaHudson (**UX**)
@andrewkfiedler 

#### Ask 2 committers to review/merge the PR and tag them here.
@AzGoalie
@djblue

#### How should this be tested?
Try using a few of the dropdowns in Intrigue as found in Advanced Search, Upload Editor, Metacard editing, etc.
I'm not sure if I chose good delimiters here so make sure to test with as many dropdowns as possible.

#### What are the relevant tickets?
[DDF-4478 Improve filtering logic for Advanced Search](https://codice.atlassian.net/browse/DDF-4478) 
Previous Behavior: 
![Previous Behavior](https://user-images.githubusercontent.com/9013407/51849873-d77e3480-22dd-11e9-8528-ab9d93ec5fb4.png)

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
